dashboard/app/views/layouts/_footer.html.haml

```diff
@@ -1,24 +1,27 @@
+- require 'cdo/footer'
 - full_width = local_assigns[:full_width]
+- options = {ge_region: params[:ge_region]}
+- footer_contents = Cdo::Footer.get_footer_contents(options)
+- locale_options = Cdo::Footer.get_locale_options(options)
 .navbar-static-top.footer
   .container{ style: 'padding: 10px;' + (full_width ? 'margin: auto' : '') }
     .row
       .span9{style: (full_width ? 'max-width: 80%;' : '')}
         - piped_spaces = ' | '
         - privacy_updated = DCDO.get('recent_privacy_policy_update', nil)
-        - if privacy_updated
-          = link_to t('footer.privacy_updated'), CDO.code_org_url('/privacy'), style: 'color: #ffa400;'
-        - else
-          = link_to t('footer.privacy'), CDO.code_org_url('/privacy'), style: 'color: #ffa400;'
-        .dim!= piped_spaces
-        = link_to t('footer.cookie_notice'), CDO.code_org_url('/cookies')
-        .dim!= piped_spaces
-        = link_to t('footer.translate'), CDO.code_org_url('/translate')
-        .dim!= piped_spaces
-        = link_to t('landing.help_support'), 'https://support.code.org', target: '_blank', id: "support"
-        .dim!= piped_spaces
-        = link_to t('footer.store'), 'https://store.code.org'
-        .dim!= piped_spaces
-        = link_to t('footer.tos_short'), CDO.code_org_url('/tos')
+        - links = (footer_contents[:links][:studio] || []).map(&:dup)
+        - links.each_with_index do |link, index|
+          - if privacy_updated && link[:title] == 'privacy'
+            - link[:title] = 'privacy_updated'
+          - if link[:domain] == 'studio.code.org'
+            - link[:url] = CDO.studio_url(link[:url])
+          - elsif link[:domain] == 'code.org'
+            - link[:url] = CDO.code_org_url(link[:url])
+          - else
+            - link[:rel] = "noopener noreferrer nofollow"
+          = link_to t("footer.#{link[:title]}"), link[:url], link.except(:title, :url, :domain)
```

Comment:
nit: "Title" is a valid html attribute on `<a>` tags and can be used as an option with `link_to`, it's the tooltip when you hover. Instead of calling this `link[:title]` it might be a bit more clear to call it `text`.

Example from [link_to](https://apidock.com/rails/ActionView/Helpers/UrlHelper/link_to) docs:
`link_to ‘Click Me’, “example.com”, target: :_blank, title: “Click to open url in a new tab”`